<launch>
  <arg name="robot_name" default="dragonfly11"/>
  <arg name="quad_odom_topic" value="/vicon/$(arg robot_name)/odom"/>
  <arg name="payload_odom_topic" value="/vicon/pl11/odom"/>
  <arg name="nodelet_manager_name" default="nodelet_manager_control"/>
  <arg name="rviz" default="1"/>

  <arg name="rosbag_dir"    value="/home/aarow/catkin_ws/src/cooperative_cable_transport_vision/bags"/>
  <arg name="bag_file"      value="$(arg rosbag_dir)/handhold-07-27-18/test1.bag"/>

  <node pkg="rosbag"
    type="play"
    name="player"
    output="screen"
    args="$(arg bag_file)"/>


    <node pkg="viz_cctv"
      type="VizCCTV"
      name="VizCCTV"
      output="screen">
      <remap from="~payload_odom_topic" to="$(arg payload_odom_topic)"/>
      <remap from="~quad_odom_topic" to="$(arg quad_odom_topic)"/>
      <remap from="~q_i_topic"      to="/control_cctv/q_i"/>
      <remap from="~q_i_dot_topic"  to="/control_cctv/q_i_dot"/>
      <remap from="~w_i_topic"      to="/control_cctv/w_i"/>
      <remap from="~u_i_topic"      to="/control_cctv/u_i"/>
      <remap from="~u_i_prl_topic"  to="/control_cctv/u_i_prl"/>
      <remap from="~u_i_prp_topic"  to="/control_cctv/u_i_prp"/>
      <remap from="~F_0_des_topic"  to="/control_cctv/F_0_des"/>
      <remap from="~M_0_des_topic"  to="/control_cctv/M_0_des"/>
      <remap from="~q_i_des_topic"  to="/control_cctv/q_i_des"/>
      <param name="fixed_frame_id" value="mocap"/>
      <rosparam file="$(find control_cctv)/config/1robot_payload_params.yaml"/>
    </node>

    <node pkg="mesh_visualization"
      type="mesh_visualization"
      name="mesh_visualization"
      output="screen">
      <param name="mesh_resource" value="package://mesh_visualization/mesh/hummingbird.mesh"/>
      <remap from="~input" to="$(arg quad_odom_topic)"/>
    </node>

    <!--nodelet manager
      -->
    <node pkg="nodelet"
      type="nodelet"
      name="$(arg nodelet_manager_name)"
      args="manager"
      output="screen"/>

    <node pkg="nodelet"
      type="nodelet"
      args="load control_cctv/NodeletControlCCTV  $(arg nodelet_manager_name)"
      name="control_cctv"
      required="true"
      output="screen">
      <rosparam file="$(find control_cctv)/config/so3_gains_dragonfly.yaml"/>
      <rosparam file="$(find control_cctv)/config/cctv_gains_dragonfly.yaml"/>
      <rosparam file="$(find control_cctv)/config/1robot_payload_params.yaml"/>
      <remap from="~quad_odom" to="$(arg quad_odom_topic)"/>
      <remap from="~payload_odom" to="$(arg payload_odom_topic)"/>
      <remap from="~position_cmd" to="position_cmd"/>
      <remap from="~so3_cmd" to="so3_cmd"/>
      <remap from="~motors" to="motors"/>
    </node>

    <node pkg="rviz"
     type="rviz"
     name="rviz"
     args="-d $(find control_cctv)/config/rviz_config.rviz"
     if="$(arg rviz)"/>


</launch>
